assets/magnify.js

```diff
@@ -1,21 +1,42 @@
 // create a container and set the full-size image as its background
 function createOverlay(image) {
+  const overlayImage = document.createElement('img');
+  overlayImage.setAttribute('src', `${image.src}`);
   overlay = document.createElement('div');
-  overlay.setAttribute('class', 'image-magnify-full-size');
-  overlay.setAttribute('aria-hidden', 'true');
-  overlay.style.backgroundImage = `url('${image.src}')`;
-  image.parentElement.insertBefore(overlay, image);
+  prepareOverlay(overlay, overlayImage);
+
+  image.style.opacity = '50%';
+  toggleLoadingSpinner(image);
+
+  overlayImage.onload = () => {
+    toggleLoadingSpinner(image);
+    image.parentElement.insertBefore(overlay, image);
+    image.style.opacity = '100%';
+  }
+
   return overlay;
 };
 
+function prepareOverlay(container, image) {
+  container.setAttribute('class', 'image-magnify-full-size');
+  container.setAttribute('aria-hidden', 'true');
+  container.style.backgroundImage = `url('${image.src}')`;
+  container.style.backgroundColor = 'var(--gradient-background)';
+}
+
+function toggleLoadingSpinner(image) {
+  const loadingSpinner = document.getElementById(`loading-spinner-${image.dataset.mediaId}`);
```

Comment:
Great catch! I've not committed the suggestion because it'll have to be `.parentElement.parentElement`, but otherwise the strategy works well, thank you for this 🎉